### DISCUSSION TITLE
Interactively selecting scale type [linear, log, ...?

### DISCUSSION THREAD

**Discussion by jw5** [vega/vega-lite#8502] [Q&A]
I'd like to be able to control the scale type [linear, log, etc] on many of my graphs. Unfortunately, I can't seem to find a way to map an interactive parameter value into the scale type encoding. I frequently deal with data sets with wide dynamic ranges. Being able to selectively use different combinations of scalings on the x and y axis is helpful for identifying underlying relationships.

Yes, I could just use a linear scale and perform a log transform before plotting. That is the hack I currently employ but then the user must mentally convert the values displayed along the axis into original unscaled values. It is marginally acceptable for color scales. But a poor user experience for x and y.

Here is an example of what I'd like to achieve (although this example is non-functional): [Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6ALQAbZskoBWOgCtY3AHaQANOCgATZAgBOjQnh4qckAIJgE6Asl2Ex3PJcTcA7oyXUwL3bGSE3WzB0JU0wVEYxHTBCINoxByVYMAAzbl0wCnR9blZk+GzYOjUNSE1rTBxgSFZdMWwtCtIC33lFFQBfdQhIDF0AawbIAI88Ep7kJXhuTQ9qBtAIHoAPBY0lqBTGZDFNIYAJdL8A11txjag8AE9YoYBHVhCjPGtGKnONyCsohchr29MDnmHTA5DAh1cYDw3DA1GQThYjGS0Kh6H6yGCyyRAH1vtF0MiUJl0GJWMgAPzrMBdKmQK5rC6bba7IYAWUiOmxsV02IA4iTEh8ln8bsh7o8lM9Xu9uhcvgUflURQCoEDICCwRCoTC4QjEEjtaj0ZicXjkgSoUSsqSKVSOhoaT1CNl0KhYA0ANpUxZypSusWmdBY2C4hU6IU9a1koYSJTIbIRqDsDx7KpUz4eQisMamPxReBjWWMyDcQzGd04D2QWPx3RqKCwO66Quq7jzVQNq6oNUAXXT1LtGj7HSAA)

Thanks,
Jim

**Comment by domoritz** [comment]
You cannot dynamically change scale types in Vega. You could map a parameter of a scale (like the exponent parameter) to a parameter/signal, though.

**Comment by jw5** [comment]
Is there an easy way to move it to a feature request?

For report generation I'm willing to just double the number of charts. But
for interactive use, keeping things tight within a single page and
responding to requests is preferred.

Thanks,
Jim

**Reply by domoritz** [reply]
It may have been requested before. The problem is that we cannot switch between continuous and discrete scales at all so it would be tricky to implement this feature. You can file a new request in Vega, though.

**Comment by jw5** [comment]
Thanks. I would have thought that the continuous/discrete scale selection
was affected by datum type not scaling function? I'll look into filing a
Vega feature request.

Your quick and informative responses are much appreciated.
